frontend/web/src/app/components/order/order-edit/order-edit.component.ts

```diff
@@ -0,0 +1,221 @@
+              const id = params['id'];
+              if (id === 'create') { // new product is being created
+                this.create();
+              } else if (id) {
+                this.getData(id);
+              }
+            });
+
+        },
+      );
+    });
+  }
+
+  create() {
+    console.log("creating");
+    /* init with a boilerplate */
+    this.creating = true;
+    if (this.creating) {
+      this.order = new Order({});
+      this.order.orderedParts = [];
+    }
+    this.populate();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (this.creating) {
+      this.create();
+    } else if (this.order) {
+      this.populate();
+    }
+  }
+
+  /* get the product */
+  getData(id) {
+    this.orderService.getOrder(id).subscribe(
+      (order) => {
+        this.order = order;
+        this.populate();
+      },
+    );
+  }
+
+  /* populateDigitlParts data */
+  private populate() {
+    this.constructForms();
+  }
+
+  /* init forms */
+  private constructForms() {
+    this.constructOrderedPartsForms();
+    const fields = {
+      customerID: [this.order && this.order.customerID ? this.order.customerID : '',
+        Validators.compose([Validators.required])],
+      date: [this.order && this.order.date ? this.order.date : '',
+        Validators.compose([Validators.required])],
+    };
+
+    this.requiredFieldsForm = this.formBuilder.group(fields);
+    this.loaded = true;
+  }
+
+  private constructOrderedPartsForms() {
+    /* load attr with default data from product instance*/
+    this.orderedPartsFieldsForm = [];
+
+    this.order.orderedParts.forEach((orderedPart) => {
+      this.orderedPartsFieldsForm.push(this.formBuilder.group({
+        amount: [orderedPart.amount, Validators.compose([Validators.required])],
+        digitalPartID: [orderedPart.digitalPartID, Validators.compose([Validators.required])],
+      }));
+    });
+  }
+
+  /* save product instance */
+  save() {
+    /* convert relevant fields */
+
+    const id = this.order.id;
+    const order: Order = new Order(this.requiredFieldsForm.value);
+    order.date = this.requiredFieldsForm.get('date').value;
+
+    this.creating ? delete order['id'] : order.id = id;
+    const orderedParts: OrderedPart[] = [];
+
+    this.orderedPartsFieldsForm.forEach((formGroup) => {
+      orderedParts.push(new OrderedPart({
+        digitalPartID: formGroup.value.digitalPartID,
+        amount: formGroup.value.amount,
+      }));
+    });
+    order.orderedParts = orderedParts;
+    if (this.creating) { // a new product
+      this.orderService.createOrder(order).subscribe(
+        (data) => {
+          if (this.nav) {
+            this.back();
+          }
+          this.changed.emit(data);
+        }, (error) => {
+          console.log(error);
+          this.errorService.showAlert(error.verobose_message_header, error.verbose_message);
+        },
+      );
+    } else {
+      this.orderService.updateOrder(order).subscribe(
+        (data) => {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this function so that its implementation is not identical to the one on line 150. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4144)